src/SymbolTable.ts

```diff
@@ -157,4 +157,4 @@ export interface BscSymbol {
 /**
  * A function that returns a symbol table.
  */
-export type SymbolTableProvider = () => SymbolTable;
+export type SymbolTableProvider = () => SymbolTable | undefined;
```

Comment:
Other parts of the program pass functions which may return `undefined` as `SymbolTableProvider`s. The place where SymbolTableProviders are read is line 43; this line will return `undefined` whether the `SymbolTableProvider` is `undefined` or whether it is a function which returns `undefined`.